wasatch/MockUSBDevice.py

```diff
@@ -0,0 +1,312 @@
+        pass
+
+    def claim_interface(self, *args, **kwargs):
+        # connecting
+        return True
+
+    def release_interface(self):
+        # disconnecting
+        return True
+
+    def ctrl_transfer(self, *args, **kwargs):
+        device, host, bRequest, wValue, wIndex, wLength = args
+        log.info(f"Mock spec received ctrl transfer of host {host}, request {bRequest}, wValue {wValue}, wIndex {wIndex}, len {wLength}")
+        if bRequest == 255:
+            cmd_func = self.cmd_dict.get((bRequest,wValue),None)
+        else:
+            cmd_func = self.cmd_dict.get((bRequest,None),None)
+        if cmd_func:
+            return cmd_func(*args)
+        else:
+            return self.default_ctrl_return
+
+    def cmd_read_eeprom(self, *args):
+        device, host, bRequest, wValue, wIndex, wLength = args
+        page = wIndex
+        return self.eeprom_obj.write_buffers[page]
+
+    def cmd_set_int_time(self, *args):
+        device, host, bRequest, wValue, wIndex, wLength = args
+        if not self.got_start_int: self.got_start_int = True
+        self.set_int_time(wIndex << 8 | wValue)
+        return [1]
+
+    def cmd_get_laser_enabled(self, *args):
+        device, host, bRequest, wValue, wIndex, wLength = args
+        return [int(self.laser_enable)]
+
+    def cmd_get_detector_temp(self, *args):
+        bytes = struct.pack('>e',self.detector_temp_raw)
+        value = int.from_bytes(bytes,byteorder='big')
+        value = value & 0x0F
+        return value.to_bytes(2, byteorder='big')
+
+
+    def cmd_toggle_laser(self, *args):
+        device, host, bRequest, wValue, wIndex, wLength = args
+        self.laser_enable = bool(wValue)
+        return [int(self.laser_enable)]
+
+    def cmd_set_gain(self, *args):
+        device, host, bRequest, wValue, wIndex, wLength = args
+        if not self.got_start_detector_gain: self.got_start_detector_gain = True
+        wValB = wValue.to_bytes(2,byteorder='little')#struct.unpack('f',bytearray(wValue))
+        lsb = wValB[0] # LSB-MSB
+        msb = wValB[1]
+        raw = (msb << 8) | lsb
+
+        gain = msb + lsb / 256.0
+        self.detector_gain = gain
+        return [1]
+
+    def cmd_set_offset(self, *args):
+        device, host, bRequest, wValue, wIndex, wLength = args
+        if not self.got_start_detector_offset: self.got_start_detector_offset = True
+        self.detector_offset = wValue
+        return [1]
+
+    def cmd_set_setpoint(self, *args):
+        device, host, bRequest, wValue, wIndex, wLength = args
+        if not self.got_start_detector_setpoint: self.got_start_detector_setpoint
+        self.detector_setpoint = wValue
+        return [1]
+
+    def cmd_toggle_tec(self, *args):
+        device, host, bRequest, wValue, wIndex, wLength = args
+        self.detector_tec_enable = bool(wValue)
+
+    def cmd_get_tec_enable(self, *args):
+        device, host, bRequest, wValue, wIndex, wLength = args
+        return [bool(self.detector_tec_enable)]
```

Comment:
I would generally return these as bytes / ints rather than bools.  The actual hardware / pyusb won't return bools.